rpc/core/types/responses_commitments.go

```diff
@@ -0,0 +1,34 @@
+package coretypes
+
+import (
+	"github.com/cometbft/cometbft/crypto/merkle"
+	"github.com/cometbft/cometbft/libs/bytes"
+)
+
+// BridgeCommitmentLeaf the leaf to form a BridgeCommitment.
+type BridgeCommitmentLeaf struct {
+	Height uint64 `json:"height"`
+
+	// The ResultsHash of blocks is derived at (Height + 1) in the LastResultsHash variable in the Tendermint
+	// block header, reference:
+	// https://github.com/cometbft/cometbft/blob/719b64156aaa3cb89add29d053439060f8e420dd/proto/cometbft/types/v1/types.proto#L67
+	// Thus to reconstruct this root at Height X, you would need the transactions results from Height X - 1.
```

Comment:
Use reference based on v0.38.5

```suggestion
	// The ResultsHash of blocks is derived at (Height + 1) in the LastResultsHash variable in the Tendermint
	// block header, ref: https://github.com/cometbft/cometbft/blob/v0.38.5/proto/tendermint/types/types.proto#L66.
	// Thus, to reconstruct this root at Height X, you would need the transactions results from Height X - 1.
```